workflow/controller/controller.go

```diff
@@ -507,12 +507,18 @@ func (wfc *WorkflowController) processNextPodCleanupItem(ctx context.Context) bo
 				return err
 			}
 		case labelPodCompleted:
+			// remove the argoproj.io/finalizer finalizer from the pod
+			p, err := pods.Get(ctx, podName, metav1.GetOptions{})
```

Comment:
You can get the pod from the pod informer without any API calls.

Reply:
@alexec done